gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/UnixTimestampRecursiveCopyableDataset.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gobblin.data.management.copy;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+import org.joda.time.DateTimeZone;
+import org.joda.time.LocalDate;
+import org.joda.time.LocalDateTime;
+import org.joda.time.Period;
+import org.joda.time.format.PeriodFormatter;
+import org.joda.time.format.PeriodFormatterBuilder;
+import org.testng.collections.Lists;
+
+import org.apache.gobblin.util.filters.AndPathFilter;
+
+
+/**
+ * This dataset reads files with format /path/to/dir/".*([0-9]{13})-PT-.*"*
+ * where the regex can be supplied through configuration.
+ * It extracts the timestamp from the path and filters paths which lie within the {@link #lookbackPeriod}
+ * It then applies the {@link #snapshotSelectionPolicy} to filter further.
+ */
+public class UnixTimestampRecursiveCopyableDataset extends RecursiveCopyableDataset {
+
+  private static final String CONFIG_PREFIX = CopyConfiguration.COPY_PREFIX + ".recursive";
+  public static final String SNAPSHOT_SELECTION_POLICY = CONFIG_PREFIX + ".snapshot.selection.policy";
+  public static final String TIMESTAMP_REGEEX = CONFIG_PREFIX + ".timestamp.pattern";
+  public static final String DEFAULT_TIMESTAMP_REGEEX = ".*([0-9]{13})-PT-.*";
```

Comment:
typo: DEFAULT_TIMESTAMP_REGEX?

Reply:
Shouldn't the regex be derived based on config for gobblin.dataset.pattern? IN other words, we should be looking for timestamp after the dataset root.